### ISSUE TITLE
Provide mechanism to use example objects

### ISSUE THREAD

**Issue by thegagne** [microsoft/typespec#5308]
### Clear and concise description of the problem

When defining complex models with full fledged examples, it would be helpful to be able to define an example object, and then reference it both as an object and as parts to fill out complete OpenAPI examples.

Example:

```tsp
const exampleWidget = #{
  id: "1",
  weight: 2,
  color: "red",
};

@example(exampleWidget)
model Widget {
  @visibility("read", "update")
  @path
  @example(exampleWidget.id)
  id: string;

  @example(exampleWidget.weight)
  weight: int32;

  @example(exampleWidget.color)
  color: "red" | "blue";
}
```

Alternatively, it would be fantastic if the model example would automatically add the doc for all the members without requiring duplicate @example tags. In most cases I would expect the same values inside the @example for the object and the @example for individual fields.

To make this more clear, it would be great if I could just define:

```tsp
@example(exampleWidget)
model Widget {
  @visibility("read", "update")
  @path
  id: string;

  weight: int32;

  color: "red" | "blue";
}
```

And have it _also_ fill in the examples for the id, weight, and color without me having to add those examples explicitly.

### Checklist

- [x] Follow our [Code of Conduct](https://github.com/microsoft/typespec/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [docs](https://typespec.io/docs/).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by markcowl** [comment]
Referencing the parts of complex examples are specifically out of scope in the value proposal (#2046), for one thing, they introduce issues for optional properties referenced in an example (what if you reference an optional example property valid in the type, but not provided in the example).

Adding to the backlog as a future design issue.

**Comment by thegagne** [comment]
I agree this could get complex, and you know better than I what is involved.

A couple more thoughts:

>they introduce issues for optional properties referenced in an example (what if you reference an optional example property valid in the type, but not provided in the example).

Are you speaking of the alternative approach where it auto fills in the property examples from the object example? I agree that this would need to be thought out. I think it's very easy to say that in the situation where an example property is not provided, you would omit an example for that property. 

Another possibility is to leverage the ability to provide multiple example, and looping through all of them to find all the fields, but this could get complex very quickly.

I don't see how this would be a problem for the original approach of just manually referencing them by example.property, since you would know if that property has been defined in the example or not.
